arch/arm64/src/imx9/imx9_usbdev.c

```diff
@@ -1870,8 +1868,16 @@ bool imx9_epcomplete(struct imx9_usb_s *priv, uint8_t epphy)
 
   up_invalidate_dcache((uintptr_t)dtd,
                        (uintptr_t)dtd + sizeof(struct imx9_dtd_s));
+
   up_invalidate_dcache((uintptr_t)dtd->buffer0,
-                       (uintptr_t)dtd->buffer0 + dtd->xfer_len);
+    (uintptr_t)dtd->buffer0 + CACHE_ALIGN_UP(dtd->xfer_len));
+
+  up_invalidate_dcache((uintptr_t)dtd->buffer1,
```

Comment:
These are most likely wrong; should just invalidate starting from buffer0. The buffers are consecutive, i.e. if for example xfer_len is larger than buffer0, it will just invalidate the correct part of buffer1 as well...